packages/core/src/modules/non-secret/service/NonSecretRecordService.ts

```diff
@@ -0,0 +1,40 @@
+import type { ConnectionRecord } from '../../connections/repository/ConnectionRecord'
+import type { NonSecretRecordTags } from '../repository/NonSecretRecord'
+
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { AriesFrameworkError } from '../../../error'
+import { NonSecretRecord } from '../repository/NonSecretRecord'
+import { NonSecretRecordRepository } from '../repository/NonSecretRecordRepository'
+
+@scoped(Lifecycle.ContainerScoped)
+export class NonSecretRecordService {
+  private nonSecretRepository: NonSecretRecordRepository
+
+  public constructor(nonSecretRepository: NonSecretRecordRepository) {
+    this.nonSecretRepository = nonSecretRepository
+  }
+
+  public async saveRecord(message: string, tags?: NonSecretRecordTags, connectionRecord?: ConnectionRecord) {
```

Comment:
I really like that we can assign custom tags to each record, so they can be efficiently searchable!

However I think it would be nice if there is an API at module/service level to **update** tags and contents for existing records. And also to **delete** them.

Reply:
Agreed! We should probably add some methods to the public api to update and delete records